<launch>

    <arg name="markerId"        default="582"/>
    <arg name="markerSize"      default="0.1"/>    <!-- in m -->
    <!--arg name="eye"             default="left"/-->
    <arg name="marker_frame"    default="aruco_marker_frame"/>
    <arg name="ref_frame"       default="camera_rgb_frame"/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->


    <node pkg="aruco_ros2" exec="single" name="aruco_single" output="screen"  >
        <remap from="/camera_info" to="/camera/color/camera_info"/>
        <remap from="/image" to="/camera/color/image_raw" /> 
        <param name="image_is_rectified" type="bool"    value="True"/>
        <param name="marker_size"    type="float"    value="$(var markerSize)"/>
        <param name="marker_id"         type="int" value="$(var markerId)"/> 
        <param name="reference_frame"    type="str"    value="$(var ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <!--param name="camera_frame"       value="stereo_gazebo_$(var eye)_camera_optical_frame"/>
        <param name="marker_frame"       value="$(var marker_frame)" />
        <param name="corner_refinement"  value="$(var corner_refinement)" /-->
    </node>

</launch>

